CONTRIBUTING.md

```diff
@@ -0,0 +1,131 @@
+so we are able to reproduce the issue.
+
+## Coding Style guidelines
+
+* Use four spaces for indentation, please don't use tabs except where
+required, such as in Makefiles.
+
+* Lines should be at most 78 chars.
+
+* Please don't leave trailing whitespace at the ends of lines.
+
+* Braces open on the same line as the `for`/`while`/`if`/`else`
+/`switch`/etc. Closing braces are put on a line of their own, except
+in the `else` of an `if` statement or before a `while` of a `do`-`while`
+statement. Always use braces for `if` and `while`.
+
+```c
+if (a == b) {
+    ...
+}
+
+if (a == b) {
+    ...
+} else if (a > b) {
+    ...
+}
+
+if (a == b) {
+    do_something();
+} else {
+    do_something_else();
+}
+
+do {
+    do_something();
+} while (cond);
+
+while (cond) {
+    do_something();   
+}
+
+switch (expression) {
+case constant-expression:
+    ...
+case constant-expression:
+    ...
+default:
+    ...
+}
+
+```
+
+* In general, do not use `goto`. The goto statement only comes in handy
+when a function exits from multiple locations and some common work such
+as cleanup has to be done. If there is no cleanup needed then just
+return directly.
+
+* For function, `static` and return type are put on the same line.
+Function's name is put on a line of its own.
+
+```c
+static tlog_grc
+tlog_play_run(struct tlog_errs **perrs, int *psignal)
+{
+              ...
+}
+```
+
+* For regular comments, use C style (`/*` `*/`) comments, even for one-line.
+
+* For doc-comments (Doxygen comments), use the `/**` `*/` style featured.
+
+* For wrapping (`\`) of macro, make sure to follow:
+
+```c
+#define CMP_BOOL(_name) \
+    do {                                                        \
+        if (_name != t._name##_out) {                           \
+            FAIL(#_name " %s != %s",                            \
+                 BOOL_STR(_name), BOOL_STR(t._name##_out));     \
+        }                                                       \
+    } while (0)
+```
+
+## Directories
+Tlog source is organized as follows.
+
+```
+.                        <- Top-level build and documentation files.
+├── doc                  <- Documentation.
+├── include              <- C headers.
+│   ├── tlog             <- Tlog library headers.
+│   ├── tltest           <- Tlog test library headers.
+├── lib                  <- C library sources.
+│   ├── tlog             <- Tlog library sources.
+│   ├── tltest           <- Tlog test library sources.
+├── src                  <- Program sources.
+│   ├── tlog             <- Tlog sources.
+│   ├── tltest           <- Tlog test sources.
+├── m4                   <- m4 macro sources.
+│   ├── autotools
+│   ├── tlog
+└── man                  <- Manpages.
+```
+
+Follow instructions in [README](README.md) for how to compile and run
+test code.
+
+## License
+By contributing to Tlog, you agree that your contributions will be licensed
+under the [GNU GPL](COPYING) in the root directory of this source tree.
```

Comment:
The words " in the root directory of this source tree" do not really apply to "GNU GPL", and this reads strange. Just remove them. It would be good to clarify which version of GNU GPL it is. so please change it to "GNU GPL v2 or later".